wcfsetup/install/files/lib/system/user/notification/UserNotificationHandler.class.php

```diff
@@ -239,6 +243,10 @@ public function fireEvent($eventName, $objectType, IUserNotificationObject $noti
 				],
 				'recipients' => $recipients
 			];
+
+			$parameters['notifications'] = $notifications;
+			$parameters['recipientIDs'] = $recipientIDs;
+			EventHandler::getInstance()->fireAction($this, 'createNotification', $parameters);
```

Comment:
You have the fireEvent event right above. Why is there another event required? In addition, the creation can be intercepted via the DBOAction.

Reply:
I checked this out, but in order to get the notification's message, title and link properly, I'll have to process the notifications (performance yaay).
But yes I care more about the update stackable and confirmed/removal events since there is no way at the moment.